BTrees/_base.py

```diff
@@ -113,7 +113,7 @@ def _search(self, key):
             k = keys[i]
             if k is key or k == key:
                 return i
-            if k < key:
+            if k is None or (key is not None and k < key):
                 low = i + 1
```

Comment:
Actually, the test above is wrong. :(

If both k and key are None, then k < key is false, which the expression above doesn't capture.

Let see:

     if key is not None and (k is None or k < key):

I should just change this to use cmp (or whatever we rename it to).